tests/system/Database/ForgeTest.php

```diff
@@ -0,0 +1,69 @@
+<?php namespace CodeIgniter\Database;
+
+use CodeIgniter\Database\MockConnection;
+
+class QueryTest extends \CIUnitTestCase
+{
+
+	protected $db;
+
+	//--------------------------------------------------------------------
+
+	public function setUp()
+	{
+		$this->db = new MockConnection([]);
+	}
+
+	//--------------------------------------------------------------------
+
+	public function testAddForeignKey()
+	{
+	    $forge = new Forge($this->db);
+	    
+	    $fk[] = [
+	        'key' => 'key',
+	        'fkey' => 'fkey',
+	        'table' => 'table',
+	        'ftable' => 'ftable',
+	        'onUpdate' => 'CASCADE',
+	        'onDelete' => 'CASCADE'
+	    ];
+	    
+	    $forge->addForeignKey('key', 'fkey', 'table', 'ftable', 'CASCADE', 'CASCADE');
+
+		$this->assertEquals($fk, $forge->getForeingKeys());
+	}
+	
+	//-------------------------------------------------------------------
+	
+	public function testCompileFkeyCreate()
+	{
+	    $forge = new Forge($this->db);
+	    
+	    $fk = [
+	            'key' => 'key',
+	            'fkey' => 'fkey',
+	            'table' => 'table',
+	            'ftable' => 'ftable',
+	            'onUpdate' => 'CASCADE',
+	            'onDelete' => 'CASCADE'
+	    ];
+	    
+	    $forge->addForeignKey('key', 'fkey', 'table', 'ftable', 'CASCADE', 'CASCADE');
+	     
+	    $sql = 'ALTER TABLE table ADD CONSTRAINT fk_fkey FOREIGN KEY (key) REFERENCES ftable (fkey) ON UPDATE CASCADE ON DELETE CASCADE';
```

Comment:
I can confirm that this should work in MS SQL and MySQL.

One thing I would like to see changed, though, would be to use the foreign table name as well as the foreign key in the name of the constraint, i.e. `fk_ftable_fkey` (obviously this would be in `Forge:: compileFKeyCreate()`, but would also have to be changed here, and I didn't want to spread multiple comments around).

It would seem fairly reasonable to need multiple foreign keys on a table which reference the `id` columns in different tables, so you would find yourself attempting to redefine the constraint `fk_id`.

Also, it looks like a few other files found their way into the PR over time.